lib/teleterm/clusters/cluster_auth.go

```diff
@@ -277,3 +281,125 @@ func (c *Cluster) processAuthResponse(ctx context.Context, key *client.Key, resp
 
 	return nil
 }
+
+// PwdlessLogin processes passwordless logins for this cluster.
+func (c *Cluster) PwdlessLogin(ctx context.Context, stream api.TerminalService_LoginPasswordlessServer) error {
```

Comment:
> FYI, I wouldn't advise passing gRPC streams around, but I assume you are following suite from other methods.
> 
> It tends to make testing awkward, as it ties various parts of the code to gRPC's behavior.

What would be another way to solve this? The only thing which comes to my mind is accepting an interface with `Recv`/`Send`, would something like that work?

Reply:
I tend to write simpler gRPC layers than what I'm seeing here. My first instinct would be to code the stream in the "topmost" gRPC handler and (possibly) call out streaming-unaware methods to solve its stages. This way you'll have a full gRPC test for the 
"topmost" method (as we usually do) and plain Go tests for everything downstream. (Passing proto messages around is fine, protos are meant to carry data after all.)

Now, TBH, I would avoid streaming if I have a fixed number of request-response calls, like we have here - I would chain unary RPCs instead. I think streaming shines when we need an arbitrary number of server-to-client messages, but otherwise the feature isn't (wasn't?) very well supported by a number of language clients.

That's my 2c on this particular scenario, but I don't expect any changes to PR. Happy to chat about gRPC/project design/organization sometime.